src/Data/provideLocalStorageDataClass.ts

```diff
@@ -142,6 +145,34 @@ export function provideLocalStorageDataClass(
   }
 }
 
+export function searchLocalStorageDataClasses(
+  search: string,
+  blackListEntities: string[] = [],
+): Array<{ _id: string; entity: string; title: string }> {
+  const results: Array<{ _id: string; entity: string; title: string }> = []
+
+  const lowerCaseSearchTerm = search.toLowerCase()
+  const matchesSearchTerm = (value: unknown) =>
+    typeof value === 'string' &&
+    value.toLowerCase().includes(lowerCaseSearchTerm)
+
+  recordKeys.forEach(({ className: entity, recordKey }) => {
+    if (blackListEntities.includes(entity)) return
+
+    Object.entries(restoreRecord(recordKey)).forEach(([_id, item]) => {
+      if (Object.values(item).some(matchesSearchTerm)) {
+        results.push({
+          _id,
+          entity,
+          title: ensureString(item.title) || ensureString(item.keyword),
+        })
+      }
+    })
+  })
+
+  return results
```

Comment:
* [ ] FYI The results will be "sorted" by the runtime order of entities. Assuming an actual backend would sort by relevance, could we apply some sorting, e.g., by the number of matches per item?